### ISSUE TITLE
Adafruit uploading of Temp

### ISSUE THREAD

**Issue by sjude68** [Aircoookie/WLED#183]
I have used Wled to check Temperature using a DHT11 sensor.  I want to upload the values to adafruit.io and also control the WLED strip from there.

I added code to add 4-more relay to control AC devices using Alexa and MQTT.  I want to control the relays using adafruit or any web/android app as Alexa (Echo) needs to be powered ON for Alexa control and MQTT needs to be running.  I am using MQTT on a Raspberry Pi 3.

A little help to configure the files.

I used your help Aircoookie to add the Alexa config (**wled12_alexa.ino**) and MQTT (**wled17_mqtt.ino**).  But can I move the config lines to **wled06_usermod.ino** so when an uploaded version of WLED comes up, I just replace the wled06_usermod file.

I get the wd reset when I add connection to adafruit.

---WLED 0.8.4 1903272 INIT---
esp8266 2_4_2
heap 26472
LEDs inited. heap usage ~184
Load EEPROM
CSSID: My WiFi NO AP
C_NC
C_NC
C_NC
C_NC
C_NC
C_NC
C_NC
C_NC
C_NC
C_NC
C_NC
C_NC

Idle. Waiting for connect to be called...

Connected! IP address: 192.168.1.194
HTTP server started

 ets Jan  8 2013,rst cause:4, boot mode:(3,6)

wdt reset
load 0x4010f000, len 1384, room 16 
tail 8
chksum 0x2d
csum 0x2d
vbb28d4a3
~ld

---WLED 0.8.4 1903272 INIT---
esp8266 2_4_2

**Comment by sjude68** [comment]
I got Temperture  uploading to Adafruit and 4 relays working.

**Comment by Mariu86** [comment]
You have all 4 relays connected to 1 nodemcu ?

**Comment by sjude68** [comment]
Yes

**Comment by Mariu86** [comment]
Very good, :D I hope to see good news from you !

**Comment by sjude68** [comment]
@Aircoookie I want to use a DHT11 temp sensor and upload it to adafruit.io

I have seen the example code which uses the code in setup.
`  
// connect to io.adafruit.com
  Serial.print("Connecting to Adafruit IO");
  io.connect();

// wait for a connection
  while(io.status() < AIO_CONNECTED) {
    Serial.print(".");
    delay(500);
  }

  // we are connected
  Serial.println();
  Serial.println(io.statusText());
`
I used the code in usermod.ino  The while loop is only to check if AIO_CONNECTED so I omitted it because I cannot use the delay function.

**Comment by Aircoookie** [comment]
@sjude68 is it working? That check is probably needed for aio to have time to connect. I'd have an idea, can you send your usermod.ino?

**Comment by sjude68** [comment]
@Aircoookie. Its not uploading the temperature to adafruit.io dashboard.  I have attached the usermod.ino file for your reference.  Please ignore the commented lines as I was testing.  

[wled06_usermod.zip](https://github.com/Aircoookie/WLED/files/4310245/wled06_usermod.zip)

**Comment by Aircoookie** [comment]
Ok, the connection should be done in `userConnect()`, not `userSetup()`, since at the point userSetup is called, WiFi is not yet connected. Here, I modified your code a bit (not sure if it helps)

[wled06_usermod_new.zip](https://github.com/Aircoookie/WLED/files/4312227/wled06_usermod_new.zip)

**Comment by sjude68** [comment]
@Aircoookie it doesn't connect to Adafruit after connection to WiFi.  The debug log as below.

-----------
---DEBUG INFO---
Runtime: 462158
Unix time: 462
Free heap: 11800
Wifi state: 6
State time: 10000
NTP last sync: 999000000
Client IP: (IP unset)
Loops/sec: 4193

Connected! IP address: 192.168.1.170
Init STA interfaces
mDNS started
Connecting to Adafruit IODisconnected!
Access point disabled.
Connecting to Zingat...
---DEBUG INFO---
Runtime: 472162
Unix time: 472
Free heap: 12240
Wifi state: 6
State time: 10000
NTP last sync: 999000000
Client IP: (IP unset)
Loops/sec: 6607

Connected! IP address: 192.168.1.170
Init STA interfaces
mDNS started
Connecting to Adafruit IODisconnected!
Access point disabled.
Connecting to Zingat...
-----------

You are using   WiFi.begin(clientSSID, clientPass);

But in Adafruit it is using AdafruitIO_WiFi io(IO_USERNAME, IO_KEY, clientSSID, clientPass);
and then io.connect()

This is what I have noticed.  I am attaching the complete sketch for your reference.
[wled00.zip](https://github.com/Aircoookie/WLED/files/4313772/wled00.zip)

**Comment by sjude68** [comment]
@Aircoookie . I created a new sketch from the usermod.ino for testing and found the upload to adafruit.io working.  But the same in wled is not working.  Can you please look into to check where I am going wrong in the mqtt connection.
[Test_WLED.zip](https://github.com/Aircoookie/WLED/files/4330904/Test_WLED.zip)

Instead of using previous connection method, I used another option found by searching adafruit.

**Comment by sjude68** [comment]
@Aircoookie I need a little help from you regarding mqtt->publish

Mqtt publish works when using Raspberry Pi as a Mqtt server.  I found a way to send feeds from Raspberry Pi to Adafruit automatically.  I tested this using MQTT Explorer on Windows 10 by sending mqtt topic manually to Raspberry Pi and it send immediately to Adafruit.io

Help me to configure in wled06_usermod.ino 

I used the line
mqtt->publish("temp", 0, true, celsius); 
but this gives exception(28) error.
